julearn/model_selection/available_searchers.py

```diff
@@ -104,10 +148,37 @@ def register_searcher(
                 "overwrite existing searchers."
             )
     logger.info(f"Registering new searcher: {searcher_name}")
-    _available_searchers[searcher_name] = searcher
+    _available_searchers[searcher_name] = {
+        "class": searcher,
+        "params_attr": params_attr,
+    }
 
 
 def reset_searcher_register() -> None:
     """Reset the searcher register to its initial state."""
     global _available_searchers
     _available_searchers = deepcopy(_available_searchers_reset)
+
+
+def get_searcher_params_attr(searcher: Union[str, Type]) -> Optional[str]:
+    """Get the name of the attribute that holds the search space.
+
+    Parameters
+    ----------
+    searcher:
```

Comment:
Missing accepted types for the argument.

Reply:
This depends on what the user registers. There's no restriction.